### PR TITLE
Chore: Task1 add models

### PR DESCRIPTION
**Pull request by jonathanrib3**
### Task 1 - Add models and db schema to our Pokemon API broker

In this branch i implemented the database modelling and models to our app. It is possible to see the relational entity diagram [here](https://dbdiagram.io/d/Digimon-API-6511c5b2ffbf5169f076fc11).

### CODE REVIEW

File: config/initializers/inflections.rb
Comment (TheDevJoao) [inline config/initializers/inflections.rb:1]: Why was this necessary? Rails does a pretty good job pluralizing stuff.

Comment (jonathanrib3) [reply config/initializers/inflections.rb:1]: Rails was pluralizing PokedexEntry to PokedexEntries, this pattern was being used to those associations tables too. It's just a personal preference, when it comes to many-to-many associations i can understand it easily when both associated models are pluralized, like `PokedexesEntries` instead of `PokedexEntries`. But nice catch, all those other lines that pluralizes and singularizes those relations that doesn't exist are useless. One more thing to adjust!

Comment (jonathanrib3) [reply config/initializers/inflections.rb:1]: Changed on 18cc7b00d6567b3eb86c2ff12094c0d824f8cfe4

**Comment by TheDevJoao** [comment]
LGTM!

**Comment by jonathanrib3** [comment]
Thanks!
